### DISCUSSION TITLE
How to assign ref to antd input?

### DISCUSSION THREAD

**Discussion by AsuraKev** [ant-design/ant-design#49312] [Q&A]
Hi there,

I am integrating with a library react phone number input where I can have a custom input component where it needs to accept a forward ref. How do we assign the ref to Input component in typescript?

thanks

**Comment by mgcrons** [comment]
Hi @AsuraKev, 

i had a similar issue, and this was how i was able to make i work 

```
import {  Input,  InputRef } from "antd";
import React, { LegacyRef, useRef } from "react";

export default InputComponent(){

  const editInputRef = React.useRef<InputRef>();

return <>
            <Input
                className=" border-none text-[#2b3674] w-full"
                ref={editInputRef as LegacyRef<InputRef>}
              />
       </>
}
```

**Comment by simon-marcus** [comment]
Hi @AsuraKev @megactek Were you able to solve this? I am still struggling using that same package `'react-phone-number-input'`

I thought I had correctly set the forward refs, but perhaps not?

Here's my code:

```tsx
import {  Input,  InputRef } from "antd";
import React, { LegacyRef, useRef } from "react";
import PhoneInput, { Country } from 'react-phone-number-input'
...
    const CustomInput = forwardRef((props, ref) => {
        return (
            <Input ref={ref as LegacyRef<InputRef>} {...props} />
        )
    });

...

return (
<PhoneInput
  inputComponent={CustomInput}
  country={country}
  defaultCountry={country}
  international
  countryCallingCodeEditable={false}
  value={phone}
  onChange={(phone) => setPhone(phone ?? '')}
/>
)

...
```

And  it does render the AntD input:
<img width="853" alt="image" src="https://github.com/user-attachments/assets/37149f74-06db-483a-a494-16cacf0dc2bf">

But when I try enter any numbers into the phone number input, nothing enters and I get these errors:
```
TypeError: element.hasAttribute is not a function
TypeError: Cannot read properties of undefined (reading 'length')
```
at `node_modules/input-format/modules/dom.js (2:1)`

**Comment by mcmxc** [comment]
Try this - https://gist.github.com/mcmxc/c43d9a2967e56fde5b76332fdc1fa83d?permalink_comment_id=5220561#gistcomment-5220561
